src/psyclone/psyir/backend/fortran.py

```diff
@@ -109,6 +109,79 @@ def gen_dims(symbol):
     return dims
 
 
+def gen_datatype(symbol):
+    '''Given a Symbol instance as input, return the datatype of the
```

Comment:
Please include the name of the Symbol in the error messages produced by this routine.

Reply:
Also, this routine feels a bit long. Is it worth creating some internal routines that handle e.g. precision=int, precision=Symbol separately?

Reply:
Symbol names added and I also forgot to document the exceptions so have done this too. I've not yet looked at splitting the function.

Reply:
I've looked at splitting the routine and don't think it would help to be honest.